src/app/components/home/home.component.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * // Copyright 2021 Omar Hoblos
+ * //
+ * // Licensed under the Apache License, Version 2.0 (the "License");
+ * // you may not use this file except in compliance with the License.
+ * // You may obtain a copy of the License at
+ * //
+ * //     http://www.apache.org/licenses/LICENSE-2.0
+ * //
+ * // Unless required by applicable law or agreed to in writing, software
+ * // distributed under the License is distributed on an "AS IS" BASIS,
+ * // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * // See the License for the specific language governing permissions and
+ * // limitations under the License.
+ */
+
+import { Component, OnInit, AfterViewInit } from '@angular/core';
+import { UtilService } from '@app/service/util.service';
+import { OidcSecurityService } from 'angular-auth-oidc-client';
+import * as dayjs from 'dayjs'
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.css']
+})
+export class HomeComponent implements OnInit, AfterViewInit {
+
+  tokenObjectForDisplay = {
+    accessToken: 'No Token Found',
+    idToken: 'No Token Found',
+    claims: {},
+    decodedAccessToken: null,
+    expirationDate: null
+  };
+
+  protected CURRENT_DATE = dayjs();
+  expirationMinutes: number;
+
+  constructor(
+    private utilService: UtilService,
+    private oidcSecurityService: OidcSecurityService
+  ) { }
+
+  ngOnInit(): void {
+    this.oidcSecurityService.checkAuth().subscribe(({ isAuthenticated, userData, accessToken, idToken, errorMessage }) => {
+      this.checkIfTokenIsInSession(accessToken, idToken)
+      if (errorMessage) {
+        console.log(errorMessage)
+      }
+    });
+  }
+
+  ngAfterViewInit(): void {
+    if (this.expirationMinutes >= 0) {
+      this.checkExpiration()
+    }
+  }
+
+  returnTokenStatus() {
```

Comment:
i believe this is not used.

Reply:
This is still being used in the template for some hide/ show logic